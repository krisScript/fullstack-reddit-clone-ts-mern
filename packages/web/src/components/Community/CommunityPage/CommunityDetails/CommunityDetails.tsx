import React, { FC, CSSProperties, useContext, useState } from 'react';
import Community from '@rddt/common/types/Community';
import PopulatedPost from '@rddt/common/types/PopulatedPost';
import { Card, Typography, Avatar, Button, Icon } from 'antd';
import RootStoreContext from 'stores/RootStore/RootStore';
import AuthState from 'types/AuthState';
import { observer } from 'mobx-react-lite';
import axios from 'axios';
import { Link } from 'react-router-dom';
import openNotification from 'utilities/openNotification';
const { Title, Text } = Typography;
interface CommunityDetailsProps {
  community: Community;
  authState: AuthState;
  addPostComponentHandler: (post: PopulatedPost) => void;
}
const CommunityDetails: FC<CommunityDetailsProps> = observer(
  ({ community, authState, addPostComponentHandler }) => {
    const { authStore, modalStore } = useContext(RootStoreContext);
    const { userId } = authStore.authState.user;
    const { name, description, subscribers, theme, _id } = community;
    console.log(userId, community.user);
    const buttonStyle: CSSProperties = {
      fontWeight: 'bold',
      background: theme.colors.base,
      color: '#ffffff',
    };
    const { isAuth } = authState;
    return (
      <>
        <Card
          title={
            <>
              Community Details{'   '}
              {userId === community.user ? (
                <Link
                  style={{ color: '#ffffff', marginLeft: '0.3rem' }}
                  to={`/communities/${_id}/mod`}
                >
                  <Icon
                    type="setting"
                    theme="filled"
                    style={{ padding: '0.2rem' }}
                  />
                  Mod Tools
                </Link>
              ) : (
                ''
              )}
            </>
          }
          headStyle={{
            background: `${theme.colors.base}`,
            color: '#ffffff',
            padding: '0',
          }}
          style={{ width: '100%', textAlign: 'center' }}
        >
          <div />
          <Avatar
            size="large"
            src={`http://localhost:8080/images/${theme.icon}`}
            alt={`${name} icon`}
          />
          <Title level={3}>{name}</Title>
          <Text>{`${subscribers} members`}</Text>
          <br />
          <Text>{`${description}`}</Text>
          <div style={{ marginTop: '1rem' }}>
            <Button
              type="primary"
              style={{ marginBottom: '0.5rem', ...buttonStyle }}
              block
            >
              JOIN
            </Button>
            <Link to={`/communities/${community._id}/posts`}>
              <Button type="primary" style={{ ...buttonStyle }} block>
                CREATE POST
              </Button>
            </Link>
          </div>
        </Card>
      </>
    );
  },
);
export default CommunityDetails;
