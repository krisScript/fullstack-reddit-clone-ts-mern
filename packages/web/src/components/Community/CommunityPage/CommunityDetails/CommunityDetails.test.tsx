import React from 'react';
import { render, fireEvent, waitForElement } from '@testing-library/react';
import CommunityDetails from './CommunityDetails';
describe('<CommunityDetails/>', () => {
  const community = {
    name: 'testName',
    subscribers: 1000,
    _id: 'testId',
    user: 'testUser',
    date: 'testDate',
    description: 'testDescription',
    theme: {
      icon: 'test.svg',
      colors: {
        base: 'baseColor',
        highlight: 'highlightColor',
      },
    },
  };
  const { container, getByText, getByAltText } = render(
    <CommunityDetails community={community} />,
  );
  const { name, subscribers } = community;
  const { icon } = community.theme;
  it('snapshot', () => {
    expect(container).toMatchSnapshot();
  });
  it(`should exist`, async () => {
    const element = getByText(name);
    expect(element).toBeTruthy();
  });
  it(`should exist`, async () => {
    const element = getByText(`${subscribers} members`);
    expect(element).toBeTruthy();
  });
  it('should exist', async () => {
    const element = getByAltText(`${name} icon`) as HTMLImageElement;
    expect(element).toBeTruthy();
    expect(element.src).toMatch(`http://localhost:8080/images/${icon}`);
  });
});
