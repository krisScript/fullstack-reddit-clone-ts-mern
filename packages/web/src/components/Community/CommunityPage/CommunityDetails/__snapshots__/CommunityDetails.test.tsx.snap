// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommunityDetails/> snapshot 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered"
    style="width: 100%; text-align: center;"
  >
    <div
      class="ant-card-head"
      style="color: rgb(255, 255, 255); padding: 0px;"
    >
      <div
        class="ant-card-head-wrapper"
      >
        <div
          class="ant-card-head-title"
        >
          Community Details
             
          
        </div>
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div />
      <span
        class="ant-avatar ant-avatar-lg ant-avatar-circle ant-avatar-image"
      >
        <img
          alt="testName icon"
          src="http://localhost:8080/images/test.svg"
        />
      </span>
      <h3
        class="ant-typography"
      >
        testName
      </h3>
      <span
        class="ant-typography"
      >
        1000 members
      </span>
      <br />
      <span
        class="ant-typography"
      >
        testDescription
      </span>
      <div
        style="margin-top: 1rem;"
      >
        <button
          class="ant-btn ant-btn-primary ant-btn-block"
          style="margin-bottom: 0.5rem; font-weight: bold; color: rgb(255, 255, 255);"
          type="button"
        >
          <span>
            JOIN
          </span>
        </button>
        <button
          class="ant-btn ant-btn-primary ant-btn-block"
          style="font-weight: bold; color: rgb(255, 255, 255);"
          type="button"
        >
          <span>
            CREATE POST
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
